src/slicegpt/gpu_utils.py

```diff
@@ -25,17 +25,17 @@ def evaluate_ppl(model_adapter: ModelAdapter, testloader: DataLoader[dict[str, t
 
     start_time = time.time()
 
-    model_adapter.model.eval()
+    model.eval()
 
-    loss_fn = torch.nn.CrossEntropyLoss(reduction="none", ignore_index=model_adapter.model.config.pad_token_id)
```

Comment:
Suggestion: the `pad_token_id` is from `model` so can we exclude it from the method signature and do `loss_fn = torch.nn.CrossEntropyLoss(reduction="none", ignore_index=model.config.pad_token_id)` ? (with appropriate logic to do `ignore_index=None` if `model.config` doesn't have a `pad_token_id`)

Reply:
Ignoring padding tokens also means that it will be ignoring bos and eos tokens too since there are all the same now. I suggest removing this and explicitly providing the mask when computing the loss.

Reply:
In trying to do this, I discovered two things. config.pad_token_id will be either an index or None. Also, for the lora_model, we will need to pass in the ignore_indices separately because lora_model object does not have this information, so perhaps we need an overload.

Reply:
Updated to account for pad_token being None. Used Pavel's tests to verify no change in results. Tests pass, hence merging now.